climada/engine/forecast.py

```diff
@@ -0,0 +1,772 @@
+                                extend=extend)
+            cbar.set_label(name)
+            cbar.formatter.set_scientific(False)
+            cbar.set_ticks([0, 1000, 10000, 100000, 1000000])
+            cbar.set_ticklabels(['0', "1'000", "10'000", "100'000", "1'000'000"])
+            plt.figtext(0.125, 0.84, tit['model_text'], fontsize='x-large', color='k', ha='left')
+            plt.figtext(0.125, 0.81, tit['explain_text'], fontsize='x-large', color='k', ha='left')
+            plt.figtext(0.9, 0.84, tit['event_day'], fontsize='x-large', color='r', ha='right')
+            plt.figtext(0.9, 0.81, tit['run_start'], fontsize='x-large', color='k', ha='right')
+            plt.subplots_adjust(top=0.8)
+            # axis.set_extent((5.70, 10.49, 45.7, 47.81), crs=ccrs.PlateCarree())
+    
+        return fig, axis
+
+    def plot_hist(self,run_datetime = None, save_fig=True,close_fig=True):
+        """ plot histogram of the forecasted impacts all ensemble members
+        Parameters:
+            run_datetime (datetime.datetime, optional): select the used hazard
+                by the run_datetime
+            save_fig (bool): True default to save the figure
+            close_fig (bool): True default to close the figure
+        """
+        # select hazard with run_datetime
+        if run_datetime == None:
+            run_datetime = self.run_datetime[0]
+        haz_ind = np.argwhere(np.isin(self.run_datetime, run_datetime))[0][0]
+        # plot histogram of all runs
+        histbin_file_name = (self.summary_str(run_datetime) +
+                             '_histbin' +
+                             '.svg')
+        histbin_file_name_full = Path(FORECAST_PLOT_DIR) / histbin_file_name
+        lower_bound = np.max([np.floor(np.log10(np.max([np.min(self._impact[haz_ind].at_event), 0.1]))), 0])
+        upper_bound = np.max([np.ceil(np.log10(np.max([np.max(self._impact[haz_ind].at_event), 0.1]))), lower_bound + 5]) + 0.1
+        bins_log = np.arange(lower_bound, upper_bound, 0.5)
+        bins = 10 ** bins_log
+        f = plt.figure(figsize=(9, 8))
+        ax = f.add_subplot(111)
+        plt.xscale('log')
+        plt.hist([np.max([x, 1]) for x in self._impact[haz_ind].at_event],
+                 bins=bins,
+                 weights=np.ones(len(self._impact[haz_ind].at_event)) / len(self._impact[haz_ind].at_event))
+        ax.yaxis.set_major_formatter(PercentFormatter(1))
+        formatter = ScalarFormatter()
+        formatter.set_scientific(False)
+        #                        formatter.format = "%:'.0f"
+        ax.xaxis.set_major_formatter(formatter)
+    
+        x_ticklabels = list()
+        x_ticks = list()
+        ticklabel_dict = {1: "1",
+                          10: "10",
+                          100: "100",
+                          1000: "1'000",
+                          10000: "10'000",
+                          100000: "100'000",
```

Comment:
Not that the habit of putting a `'` in to separate thousanders is a Swiss-only convention. 

It doesn't even make it into the example by [oracle](https://docs.oracle.com/cd/E19455-01/806-0169/overview-9/index.html).

Also, the SI convention [\#16](https://physics.nist.gov/cuu/Units/checklist.html)  is 15 739.012 53

Reply:
This is updated